File: temp_sensor/influx.py
from influxdb import InfluxDBClient
from dotenv import load_dotenv
from os import getenv
from datetime import datetime, timedelta


'''
Connect to influxDB and return temperature readings from input time ranges

TODOs - figure out how length will get used and calculate a where clause
TODOs - possible use pandas for the DB query, return format might be better
''' 

def get_temperature_data(length):

    # Load environment variables for DB auth
    load_dotenv()

    # # establish connection
    client = InfluxDBClient(
        host = '167.71.158.245', 
        port = 8086, 
        username = getenv('INFLUX_NAME'),
        password = getenv('PASSWD'),
        database = getenv('DB'))

    
    # This will be how to determine the starting time value to pull from
    # Need to figure out how to interchange hours and days
    # Length will be hours and days will just be represented by many hours

    time = datetime.now() - timedelta(hours = 7)
    
    # Still need to determine how the length param will work in this
    # One idea is to return a pandas dataframe, but it might be easier to return an R dataframe
    # I have an issue with the time type being included in the where clause
    res = client.query( f"select temperature, humidity from room_temperature_humidity limit 5;")
    
    return list(res)

    # return client.query( f"select * from room_temperature_humidity where time >= {time};")

ret = get_temperature_data(7)
print(ret)

File: plumber/influx.R

#* @get /temp
function() {
  
  con <- influx_connection(
    host = Sys.getenv('postgre_ip'), 
    port = 8086, 
    user = Sys.getenv("influx_name"),
    pass = Sys.getenv("passwd")
  )
  
  res <- influx_select(con, 
                       db = "temperature",
                       measurement = 'room_temperature_humidity',
                       limit = 1, 
                       field_keys = 'temperature',
                       order_desc = T,
                       return_xts = F,
                       simplifyList = T)
  
  temp <- res[[1]]$temperature * (9/5) + 32
  return(temp)
}

